install.php

```diff
@@ -580,9 +580,9 @@ function process_form(the_form)
 	// Check if InnoDB is available
 	if ($db_type == 'mysql_innodb' || $db_type == 'mysqli_innodb')
 	{
-		$result = $db->query('SHOW VARIABLES LIKE \'have_innodb\'');
-		list (, $result) = $db->fetch_row($result);
-		if ((strtoupper($result) != 'YES'))
+		$result = $db->query('SELECT SUPPORT FROM INFORMATION_SCHEMA.ENGINES WHERE ENGINE=\'InnoDB\'');
+		list ($result) = $db->fetch_row($result);
```

Comment:
Do we even need the `list()` call here? You could probably use `fetch_result` instead.